From e5af72acd284db1ee2b79aeeb06714770d1ad5c3 Mon Sep 17 00:00:00 2001
From: Diogo Pereira <sir.suriv@gmail.com>
Date: Thu, 20 Aug 2015 15:05:25 +0100
Subject: [PATCH] Remove -Werror, -pedantic and -g from CFLAGS

Based on upstream commits:
0b619fa build: -Werror and -pedantic in debug mode only
856c260 build: remove redundant -Werror and -pedantic

https://github.com/andrewrk/libgroove/compare/44aa5db96cfe45e56fa1f7d762fb032ca178a35c...856c26013e4356d4fc078da980083bc4d002771d
---
 CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a1e8541..14aca44 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,8 +135,11 @@ configure_file (
   "${PROJECT_BINARY_DIR}/config.h"
   )
 
-set(LIB_CFLAGS "${C99_C_FLAGS} -pedantic -Werror -Wall -Werror=strict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_REENTRANT -D_POSIX_C_SOURCE=200809L")
-set(EXAMPLE_CFLAGS "${C99_C_FLAGS} -pedantic -Werror -Wall -g")
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Wno-unused-variable")
+set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Werror -pedantic")
+
+set(LIB_CFLAGS "-std=c99 -Wall -Werror=strict-prototypes -Werror=old-style-definition -Werror=missing-prototypes -D_REENTRANT -D_POSIX_C_SOURCE=200809L")
+set(EXAMPLE_CFLAGS "-std=c99 -Wall")
 set(EXAMPLE_INCLUDES "${PROJECT_SOURCE_DIR}")
 
 add_library(groove SHARED ${LIBGROOVE_SOURCES} ${LIBGROOVE_HEADERS})
@@ -306,6 +309,7 @@ message("\n"
 "Installation Summary\n"
 "--------------------\n"
 "* Install Directory            : ${CMAKE_INSTALL_PREFIX}\n"
+"* Build Type                   : ${CMAKE_BUILD_TYPE}\n"
 "* Build libgroove              : ${LIBGROOVE_STATUS}\n"
 "* Build libgrooveplayer        : ${LIBGROOVE_PLAYER_STATUS}\n"
 "* Build libgrooveloudness      : ${LIBGROOVE_LOUDNESS_STATUS}\n"
-- 
2.5.0

